app/shop.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from app.customer import Customer
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict
+
+    def calculate_product_cost(self, customer: Customer) -> float:
+        total_price = 0
+        for product, quantity in customer.product_cart.items():
+            if product in self.products:
+                total_price += quantity * self.products[product]
+
+        return total_price
+
+    def print_purchase_receipt(self, customer: Customer) -> None:
+        actual_dt = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+        total_price = self.calculate_product_cost(customer)
+
+        print("\nDate:", actual_dt)
+        print(f"Thanks, {customer.name}, for your purchase!\nYou have bought:")
+
+        for product, quantity in customer.product_cart.items():
+            price = quantity * self.products[product]
+            if price == int(price):
```

Comment:
use isinteger

Reply:
I can`t do that, as i know there is know isinteger in Python, and when i try to use isinstance it gives the wrong result, because 3.0 for example is not an integer, but i need it to be an integer, I only can check if string of this number after . is 0, but it`ll make my code more complex. Maybe you know some other way how to do it?

Reply:
Thank you!